### ISSUE TITLE
Partial code colorization when after fixing an error

### ISSUE THREAD

**Issue by lgacnik** [microsoft/vscode-cpptools#11297]
### Environment

- OS and Version: Windows 10 Pro, 22H2
- VS Code Version: 1.81.0 (user-setup)
- C/C++ Extension Version: v2.0.0

### Bug Summary and Steps to Reproduce

Bug Summary:
I'm not sure whether the "C/C++ Themes" extension is to blame but currently I'm not using any other C/C++ colouring scheme besides the VS Code default built-in one. The issue is that after I make some code error and later fix it, syntax colouring is only partially applied to the code without errors.

Steps to reproduce:
I can only post pictures otherwise it won't be clear where I'm getting at.

**Case of error (e.g. when `enum class` values are provided but no scope specifier given)**
<img width="341" alt="2023-08-06 15_26_52-i2c_interface cpp - HMS_v0 1 - Visual Studio Code" src="https://github.com/microsoft/vscode-cpptools/assets/79907158/243b03ba-5ec6-4439-ad76-86a9c4e80095">

**Case of fixed error with partial colourization (enum values remain uncoloured except the top enum from other class which wasn't modified to begin with)**
<img width="421" alt="2023-08-06 15_27_27-i2c_interface cpp - HMS_v0 1 - Visual Studio Code" src="https://github.com/microsoft/vscode-cpptools/assets/79907158/384607c6-3f15-44d5-bedc-efbd2acee658">

Expected behavior:
**Case of fixed error with full colourization (after VS Code is restarted)**
<img width="416" alt="2023-08-06 15_28_52-i2c_interface cpp - HMS_v0 1 - Visual Studio Code" src="https://github.com/microsoft/vscode-cpptools/assets/79907158/167ede66-5200-4de4-80d1-f271e1e9b604">

### Configuration and Logs

```shell
There is a lot of going on inside C/C++ logs so I won't put it here for now unless someone requests it for a specific behaviour.

The following is `c_cpp_properties.json` file I use in my C++ project:


{
    "version": 4,
    "configurations": [
        {
            "name": "STM32",
            "configurationProvider": "ms-vscode.cmake-tools",
            "intelliSenseMode": "${default}"
        }
    ]
}
```

### Other Extensions

I don't use other colouring extensions except all C/C++ extension.

### Additional context

There is a temporary, but manual, solution for this. In the bottom right corner there is a "C++" symbol which gives "Select language mode" if I hover over it. Upon clicking it I select "Auto mode" and all C++ syntax becomes properly coloured afterwards.

I can reproduce this error by copy-pasting large chunks of code (+100 lines) where only certain user-defined keywords remain properly coloured and the rest is of white colour.

**Comment by MucaGinger** [comment]
There is a temporary, but manual, solution for this. In the bottom right corner there is a "C++" symbol which gives "Select language mode" if I hover over it. Upon clicking it I select "Auto mode" and all C++ syntax becomes properly coloured afterwards.

I can reproduce this error by copy-pasting large chunks of code (+100 lines) where only certain user-defined keywords remain properly coloured and the rest is of white color.

**Comment by Colengms** [comment]
Hi @MucaGinger .  Can you clarify which version of the C/C++ Extension you are repro'ing this with? You mentioned 2.0, but the most recently released version of the C/C++ Extension is 1.16.3.

I'm not able to easily reproduce this issue with 1.16.3.  Could you try our latest pre-release version 1.17.0 ?  If that repros the issue, would you be able to provide a stand-alone example, such as a zip file or GitHub repo, that we could use to investigate further?
